### DISCUSSION TITLE
IR Capture with Automate Things IR Bridge IR Controller (AT-IRBR v1.4)

### DISCUSSION THREAD

**Discussion by Gongman514** [arendst/Tasmota#16155] [Q&A] [answered]
Hi all,
I am working with AT-IRBR v1.4 running Tasmota 12.0.2 (tasmota-ir.bin). I've been getting inconsistent captures that are not usable. Here is a capture of a "Guide" key. 

21:17:00.715 RSL: RESULT = {"SetOption89":"OFF"}
Attempt 1
21:17:09.137 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":44,"Hash":"0x27F35ADF","Repeat":0}}
21:17:09.388 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":16,"Hash":"0xF38A2BCD","Repeat":0}}

Attempt 2 
21:18:02.218 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":4,"Hash":"0xDBC8CB72","Repeat":0}}
21:18:02.718 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":5,"Hash":"0x74FE4A6B","Repeat":0}}

Attempt 3
21:21:54.522 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":10,"Hash":"0x23E0213F","Repeat":0}}
21:21:54.924 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":45,"Hash":"0x5526197D","Repeat":0}}

21:22:43.533 RSL: RESULT = {"SetOption89":"ON"}
Attempt 1
21:22:48.826 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":51,"Hash":"0x826879C2","Repeat":0}}
21:22:49.127 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":41,"Hash":"0x1EC953C1","Repeat":0}}
21:22:49.328 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":30,"Hash":"0xB68CA14C","Repeat":0}}
21:22:49.528 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":30,"Hash":"0xA465BAFA","Repeat":0}}
21:22:49.829 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":42,"Hash":"0x21FDA14A","Repeat":0}}

Attempt 2
21:24:10.918 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":11,"Hash":"0xA6C0A4B5","Repeat":0}}
21:24:11.419 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":46,"Hash":"0x6141F312","Repeat":0}}
21:24:11.517 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":6,"Hash":"0x671A1C03","Repeat":0}}

Attempt 3
21:25:54.888 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":26,"Hash":"0xE932C826","Repeat":0}}
21:25:55.089 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":19,"Hash":"0xEFD7E85D","Repeat":0}}
21:25:55.188 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":3,"Hash":"0x24AE7D4F","Repeat":0}}
21:25:55.617 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":36,"Hash":"0xBEC649EF","Repeat":0}}

My goal is to capture keys from my remote and play them back via an MQTT message. Any help, tips, or advice are most welcome.

Thank you,

Bob

**Comment by sfromis** [comment, marked as answer]
When `UNKNOWN`, that means that the protocol is not supported, and the printed messages are not useful as is. You could try with the expanded protocol if you upgrade to `tasmota-ir.bin.gz`

You may be able to do something with raw encoding, but that's no sure-fire avenue. https://tasmota.github.io/docs/IRSend-RAW-Encoding/

If your device is some sort of HVAC unit, things are more complex than the concept of a "replay attack" with captured simple codes.

Using `SetOption89` is only relevant for Zigbee, not for IR.

**Reply by Gongman514** [reply]
Thank you for your reply @sfromis. I am using Tasmota 12.0.2 (tasmota-ir.bin).  The remote I am trying to capture is a Samsung GX-VD940CJ set-top box used by Videotron in Montreal. I will check out the RAW encoding and advise. 

Cheers

**Comment by richard-scott** [comment]
I make the AT-IRBR devices... I've tried compiling a custom firmware that enables the USE_IR_REMOTE_FULL flag, but I have a Samsung TV remote that just won't work with it.

**Reply by sfromis** [reply]
If more IR protocol support is needed, that does not happen inside the Tasmota project, as the IR decoding is done using 
https://github.com/crankyoldgit/IRremoteESP8266

**Reply by barbudor** [reply]
Also the prebuilt tasmota-ir.bin has already all known protocols enabled
